src/ray/common/status.cc

```diff
@@ -36,65 +55,83 @@ void Status::CopyFrom(const State *state) {
 
 std::string Status::CodeAsString() const {
   if (state_ == NULL) {
-    return "OK";
+    return kStatusCodeOK;
   }
 
-  const char *type;
   switch (code()) {
   case StatusCode::OK:
-    type = "OK";
-    break;
+    return kStatusCodeOK;
   case StatusCode::OutOfMemory:
-    type = "Out of memory";
-    break;
+    return kStatusCodeOutOfMemory;
   case StatusCode::KeyError:
-    type = "Key error";
-    break;
+    return kStatusCodeKeyError;
   case StatusCode::TypeError:
-    type = "Type error";
-    break;
+    return kStatusCodeTypeError;
   case StatusCode::Invalid:
-    type = "Invalid";
-    break;
+    return kStatusCodeInvalid;
   case StatusCode::IOError:
-    type = "IOError";
-    break;
+    return kStatusCodeIOError;
   case StatusCode::ObjectExists:
-    type = "ObjectExists";
-    break;
+    return kStatusCodeObjectExists;
   case StatusCode::ObjectStoreFull:
-    type = "ObjectStoreFull";
-    break;
+    return kStatusCodeObjectStoreFull;
   case StatusCode::UnknownError:
-    type = "Unknown error";
-    break;
+    return kStatusCodeUnknownError;
   case StatusCode::NotImplemented:
-    type = "NotImplemented";
-    break;
+    return kStatusCodeNotImplemented;
   case StatusCode::RedisError:
-    type = "RedisError";
-    break;
+    return kStatusCodeRedisError;
   case StatusCode::TimedOut:
-    type = "TimedOut";
-    break;
+    return kStatusCodeTimedOut;
   case StatusCode::Interrupted:
-    type = "Interrupted";
-    break;
+    return kStatusCodeInterrupted;
+  case StatusCode::IntentionalSystemExit:
+    return kStatusCodeIntentionalSystemExit;
+  case StatusCode::UnexpectedSystemExit:
+    return kStatusCodeUnexpectedSystemExit;
   default:
-    type = "Unknown";
-    break;
+    return kStatusCodeUnknown;
```

Comment:
you can define a map from code to string with macro, and then just look up the map

Reply:
fixed